### ISSUE TITLE
Magic LEF/DEF Based DRC check fails where as GDS DRC check passes

### ISSUE THREAD

**Issue by dineshannayya** [The-OpenROAD-Project/OpenLane#695]
### Description

Openlane support two way of MAGIC DRC check 1) LEF/DEF and 2) GDS using the environmental variable MAGIC_DRC_USE_GDS
I see GDS base DRC check passes, while as LEF/DEF based DRC fails with Error

### Environment
```
Python: v3.8.10
Kernel: Linux v5.11.0-38-generic
Distribution: ubuntu 20.04
Container Engine: docker v20.10.10
OpenLane Git Version: 2021.10.25_20.35.00
---
PDK Version Verification Status: OK

---
Git Log (Last 5 Commits)

commit 8d686c081c2c9aefa16dbbd8ccf5bc8f4dcabc4b
Author: Manar <manarabdelatty@aucegypt.edu>
Date:   Wed Oct 27 08:12:35 2021 -0500

    Update openroad + add multi-corner timing analysis (#680)
    
    * Update OR App
    
    * Fix overflow iterations
    
    * Fix repair_tie_fanout command
    
    * Update design config
    
    * Add multi-corner analysis

commit 683151cc9e601c2f2d774fbcc53cf23060787a41
Author: Vitor Bandeira <vvbandeira@users.noreply.github.com>
Date:   Wed Oct 27 09:36:07 2021 -0300

    Make sure docker logs dir exists (#679)
    
    Signed-off-by: Vitor Bandeira <vitor.vbandeira@gmail.com>
    
    [skip ci]

commit 354a25f7743fe68fa02f6337a922594ac41b7cee
Author: Manar <manarabdelatty@aucegypt.edu>
Date:   Tue Oct 26 09:57:47 2021 -0500

    Update resizer options (#678)
    
    * Update design configuration to address timing violations
    
    * update resizer scripts
    
    - added repair_tie_fanout command
    - added catch to the repair_timing commands to prevent exiting when the utilization limit is reached
    
    * Reset routing_cores to 2

commit c0b75f5f37717201f90396efe835e1978e32dbe6
Author: Anton Blanchard <anton@ozlabs.org>
Date:   Wed Oct 27 00:11:33 2021 +1100

    Fix typo in python exit call (#677)
    
    In commit 4e158a77ef7a ("Quit on timing violations at the typical
    corner (#659)"), os.EX_DATA should have been os.EX_DATAERR.

commit e8f4a88f668b366f126bba40861153bf478a33c1
Author: Donn <me@donn.website>
Date:   Mon Oct 25 20:32:41 2021 +0200

    Last Minute Fixes for MPW3 (#675)
```

### Reproduction Material
1. make sure that $PDK_ROOT is configure to PDK path where Sky130A installed
2. Use run command to reproduce the issue: run_cmd
[exp2.tar.gz](https://github.com/The-OpenROAD-Project/OpenLane/files/7501614/exp2.tar.gz)

3. For Checking GDS mode of DRC checking, change File:  input/drc.tcl
set ::env(MAGIC_DRC_USE_GDS) "0"
           to
set ::env(MAGIC_DRC_USE_GDS) "1"
and run the run_cmd

### Expected behavior
As GDS based DRC is passing expectation is LVS/DEF also should pass.

### Logs

```
Log and Report are part of the example tar ball under output directory

```

**Comment by RTimothyEdwards** [comment]
I would like to add that the specific problem here is that the OpenROAD router appears not to honor the wide-spacing rules between the routes and the obstruction layers defined in the macros.  There happen not to be any actual DRC errors because the obstruction layers extend out further than any of the internal routes, but that does not invalidate the issue, since the router should be maintaining a DRC distance per the information it sees in the macro LEF.

**Comment by donn** [comment]
So there are two issues here: one has been filed to Open_PDKs. The other is an issue with OpenROAD generating a bad DEF, so only you can test that.

**Comment by donn** [comment]
Issue was: https://github.com/The-OpenROAD-Project/OpenROAD/issues/2892

But fix is blocked on: https://github.com/The-OpenROAD-Project/OpenROAD/issues/3131

**Comment by kareefardi** [comment]
Closing this as blocked PR is merged. If similar issues surfaces please open a new issue.
